hydra_server/server/service.py

```diff
@@ -163,4 +165,45 @@ def login(ctx, username, password):
         login_response.user_id = user_id
         login_response.username = username
 
+        log.info(ctx.transport.req_env['beaker.session'])
+
+        return login_response
+
+    @rpc(Unicode, _returns=LoginResponse)
+    def get_remote_session(ctx, session_id=None):
+        """
+            This function takes a session id which was not generated by this server,
+            but by another hydra server. It makes a request to the remote server using
+            the endpoint 'get_hydra_session', which returns the user ID, username and session ID.
+            From here we can set the session ID as though it was generated here.
+
+        """
+        remote_hydra_server = config.get('SSO', 'url', 'localhost').strip('/')
```

Comment:
I had to add the transport to this line to get this to work.

```suggestion
        remote_hydra_server = config.get('SSO', 'url', 'http://localhost').strip('/')
```

The first attempt below failed (returning `None`) and use just `localhost` as the `remote_hydra_server`.  This gave...
```
1310735 2022-05-18 17:19:15,239 - INFO - Received request: <function AuthenticationService.get_remote_session at 0x7f112512faf0>
1310735 2022-05-18 17:19:15,241 - CRITICAL - No connection adapters were found for 'localhost:8081/get_hydra_session/97d7fb5201c24fc1b5b4a254d4b12999'
```

The second correct response is after adding `http://` as above...



```python
>>> client=RemoteJSONConnection("http://localhost:8080/json")
1310600 2022-05-18 17:10:27,603 - INFO - Using user-defined URL: http://localhost:8080/json
>>> client.login("pmslavin@gmail.com", "rfc1034")
1310600 2022-05-18 17:10:55,125 - INFO - Calling: login
1310600 2022-05-18 17:10:55,125 - INFO - Args {'login': {'username': 'pmslavin@gmail.com', 'password': 'rfc1034'}}
1310600 2022-05-18 17:10:55,691 - INFO - 97d7fb5201c24fc1b5b4a254d4b12999
1310600 2022-05-18 17:10:55,692 - INFO - done (0.5666725635528564)
1310600 2022-05-18 17:10:55,692 - INFO - Login response OK for user: 3
(3, '97d7fb5201c24fc1b5b4a254d4b12999')
>>> client.get_remote_session(session_id="97d7fb5201c24fc1b5b4a254d4b12999")
1310600 2022-05-18 17:12:14,527 - INFO - Calling: get_remote_session
1310600 2022-05-18 17:12:14,528 - INFO - Args {'get_remote_session': {'session_id': '97d7fb5201c24fc1b5b4a254d4b12999'}}
1310600 2022-05-18 17:12:14,702 - INFO - done (0.17462897300720215)
{'user_id': None}
>>> client.get_remote_session(session_id="97d7fb5201c24fc1b5b4a254d4b12999")
1310600 2022-05-18 17:15:55,951 - INFO - Calling: get_remote_session
1310600 2022-05-18 17:15:55,952 - INFO - Args {'get_remote_session': {'session_id': '97d7fb5201c24fc1b5b4a254d4b12999'}}
1310600 2022-05-18 17:15:56,138 - INFO - done (0.1869831085205078)
{'user_id': 3}
```